### PR TITLE
feat(keycloak): add seeding data to configuration

### PR DESCRIPTION
**Pull request by ntruchsess**
## Description

This new PR is opened as the original PR #800 was closed due to retirement of catenax-ng org.

The keycloak-seeder has been changed to (optionaly) read multiple json-files per realm and seeding-data from it's settings and merge this additional data with the one that is read from json before writing changes to keycloak. This allows to both separate seeding-data of different domains and override values in the keycloak-json by settings being passed as environment-variables (which can be defined via helm-chart). 
The following client-properties can now be updated: RootUrl, BaseUrl, AdminUrl, RedirectUrls, Secret. (Before the change those values could only be initially set on creation of the respective entity)
KeecloakSeederSettings have been adjusted accordingly:
```
  "KeycloakSeeding": {
    "Realms": [
      {
        "Realm": "CX-Central",
        "InstanceName": "test",
        "DataPaths": [
          "realms/CX-Central-realm.json"
        ]
       "Id": "realms internal id",
       "DisplayName" "realms display name",
       ..... further realm configuration
       "Clients": [
       ],
       "IdentityProviders": [
       ],
       .... even further realm configuration...
      }
    ]
  }
```

## Why

seeding e.g. secrets from json-files would uncover confidential data. It also is unflexible - settings defined by environment-variables can e.g. be defined in kubernetes secrets.

## Issue

#438

## Checklist

Please delete options that are not relevant.

- [X] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [X] I have performed a self-review of my own code
- [X] I have successfully tested my changes locally
- [X] I have added tests that prove my changes work
- [X] I have checked that new and existing tests pass locally with my changes
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by MaximilianHauer** [comment]
@ntruchsess could you check the quality gate ?

File: src/framework/Framework.Async/Directory.Build.props [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 <Project>
   <PropertyGroup>
-    <VersionPrefix>2.7.0</VersionPrefix>
+    <VersionPrefix>2.8.0</VersionPrefix>
```
Comment (Phil91) [inline src/framework/Framework.Async/Directory.Build.props:22]: The version needs to be updated due to an update on the dev branch

**Review by Phil91 — COMMENTED** [review]
Beside of the package version adjustment the pr looks good from my side (without testing the seeding itself)

File: src/framework/Framework.Async/Directory.Build.props (hunk 1 above)
Comment (ntruchsess) [reply src/framework/Framework.Async/Directory.Build.props]: done

**Review by evegufy — CHANGES_REQUESTED** [review]
Hi @ntruchsess I just test again and I'm encountering an unhandled expection:
`System.NullReferenceException: Object reference not set to an instance of an object.`

[logs.log](https://github.com/user-attachments/files/17029717/logs-from-centralidp-keycloak-realm-seeding-in-centralidp-keycloak-realm-seeding-2-ct522.log)

**Review by Phil91 — APPROVED** [review]

File: src/keycloak/Keycloak.Seeding/appsettings.json [hunk 2]
```diff
@@ -32,8 +32,6 @@
     }
   },
   "KeycloakSeeding": {
-    "DataPathes": [],
-    "InstanceName": "",
-    "ExcludedUserAttributes": []
+    "Realms": []
```
Comment (evegufy) [inline src/keycloak/Keycloak.Seeding/appsettings.json:35]:
```suggestion
    "Realms": [
      {
        "Realm": "",
        "InstanceName": "",
        "DataPaths": []
      }
    ]
```

File: src/keycloak/Keycloak.Seeding/BusinessLogic/SeedDataHandler.cs [hunk 3]
```diff
@@ -34,82 +35,94 @@ public class SeedDataHandler : ISeedDataHandler
...
     };
 
-    private KeycloakRealm? jsonRealm;
+    private KeycloakRealm? _keycloakRealm;
     private IReadOnlyDictionary<string, string>? _idOfClients;
 
-    public async Task Import(string path, CancellationToken cancellationToken)
+    public async Task Import(KeycloakRealmSettings realmSettings, CancellationToken cancellationToken)
     {
+        _keycloakRealm = (await realmSettings.DataPathes
```
Comment (evegufy) [inline src/keycloak/Keycloak.Seeding/BusinessLogic/SeedDataHandler.cs:43]: remove typo:
```suggestion
        _keycloakRealm = (await realmSettings.DataPaths
```

**Review by evegufy — CHANGES_REQUESTED** [review]

**Review by evegufy — CHANGES_REQUESTED** [review]
During testing, I noticed an unhandled exception with status code 500 (Internal Server Error) during the deletion of assigned default client scopes:

```
"Exception":"Org.Eclipse.TractusX.Portal.Backend.Framework.ErrorHandling.ServiceException: Internal Server Error: http://sharedidp-keycloak/auth/admin/realms/master/clients/34e015a5-b944-46d8-a9f3-4bc8e856a727/default-client-scopes/ff556da8-6b2b-4fe3-97cf-4a9c93ffc3c1\n ---> Flurl.Http.FlurlHttpException: Call failed with status code 500 (Internal Server Error): DELETE http://sharedidp-keycloak/auth/admin/realms/master/clients/34e015a5-b944-46d8-a9f3-4bc8e856a727/default-client-scopes/ff556da8-6b2b-4fe3-97cf-4a9c93ffc3c1\n
```

Scenario:

-  master-realm is created and to default clients `admin-cli`, `account` and `security-admin-console` the default client scope `acr` is assigned
- seeding job uses a json file for the master realm where the default client  scope `acr` is not assigned to default clients `admin-cli`, `account` and `security-admin-console`
- seeding job runs into the exception above BUT only after successfully removing the assignments of the default client scope `acr` from default clients `admin-cli`, `account` and `security-admin-console`

Why is this exception thrown if the deletion is successful?

File: src/keycloak/Keycloak.Seeding/BusinessLogic/SeedDataHandler.cs (hunk 3 above)
Comment (ntruchsess) [reply src/keycloak/Keycloak.Seeding/BusinessLogic/SeedDataHandler.cs:43]: Plural of path is pathes (see https://en.wiktionary.org/wiki/path )

Comment (ntruchsess) [reply src/keycloak/Keycloak.Seeding/BusinessLogic/SeedDataHandler.cs]: adjusted: https://github.com/eclipse-tractusx/portal-backend/pull/910/commits/43af89b15371c72b752b29a73723a2cde65d7a5e

File: src/keycloak/Keycloak.Seeding/appsettings.json (hunk 2 above)
Comment (ntruchsess) [reply src/keycloak/Keycloak.Seeding/appsettings.json:35]: adjusted: https://github.com/eclipse-tractusx/portal-backend/pull/910/commits/43af89b15371c72b752b29a73723a2cde65d7a5e

**Review by evegufy — APPROVED** [review]
I created https://github.com/eclipse-tractusx/portal-backend/issues/1039 to follow up on  https://github.com/eclipse-tractusx/portal-backend/pull/910#pullrequestreview-2314893821

**Pull request merged**